pkg/monitor/apiserveravailability/pod_log.go

```diff
@@ -0,0 +1,148 @@
+		if interval.From.After(end) || interval.To.Before(beginning) {
+			continue
+		}
+		timeFiltered = append(timeFiltered, interval)
+	}
+
+	sort.Sort(timeFiltered)
+	return timeFiltered, nil
+}
+
+var interestingNamespaces = []string{"openshift-cluster-version"}
+
+func SummarizeInterestingPodLogs(ctx context.Context, client kubernetes.Interface) (*APIServerClientAccessFailureSummary, error) {
+	summary := &APIServerClientAccessFailureSummary{}
+	wg := sync.WaitGroup{}
+	for _, namespace := range interestingNamespaces {
+		wg.Add(1)
+		go func(ctx context.Context, namespace string) {
+			defer wg.Done()
+
+			currSummary, err := summarizeLogsForNamespace(ctx, client, namespace)
+			if err != nil {
+				// ignore error and continue
+				klog.Warningf("error summarizing pod logs: %v", err)
+			}
+			summary.AddSummary(currSummary)
+		}(ctx, namespace)
+	}
+	wg.Wait()
+
+	return summary, nil
+}
+
+func summarizeLogsForNamespace(ctx context.Context, client kubernetes.Interface, namespaceName string) (*APIServerClientAccessFailureSummary, error) {
+	podList, err := client.CoreV1().Pods(namespaceName).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	summary := &APIServerClientAccessFailureSummary{}
+	wg := sync.WaitGroup{}
+	for i := range podList.Items {
+		pod := podList.Items[i]
+
+		wg.Add(1)
+		go func(ctx context.Context, pod *corev1.Pod) {
+			defer wg.Done()
+
+			currSummary, err := summarizeLogsForPod(ctx, client, pod)
+			if err != nil {
+				// ignore error and continue
+				klog.Warningf("error summarizing pod logs: %v", err)
+			}
+			summary.AddSummary(currSummary)
+		}(ctx, &pod)
+	}
+	wg.Wait()
+
+	return summary, nil
+}
+
+func summarizeLogsForPod(ctx context.Context, client kubernetes.Interface, pod *corev1.Pod) (*APIServerClientAccessFailureSummary, error) {
+	pod, err := client.CoreV1().Pods(pod.Namespace).Get(ctx, pod.Name, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	containerNames := []string{}
+	for _, container := range pod.Spec.InitContainers {
+		containerNames = append(containerNames, container.Name)
+	}
+	for _, container := range pod.Spec.Containers {
+		containerNames = append(containerNames, container.Name)
+	}
+
+	summary := &APIServerClientAccessFailureSummary{}
+	wg := sync.WaitGroup{}
+	for _, previousContainer := range []bool{true, false} {
+		for _, containerName := range containerNames {
+			wg.Add(1)
+			go func(ctx context.Context, previousContainer bool, containerName string) {
+				defer wg.Done()
+
+				locator := monitorapi.LocatePodContainer(pod, containerName)
+				currSummary := &APIServerClientAccessFailureSummary{}
+				logOptions := &corev1.PodLogOptions{
+					Container:                    containerName,
+					Follow:                       false,
+					Previous:                     previousContainer,
+					Timestamps:                   true,
+					InsecureSkipTLSVerifyBackend: true, // set in case the kubelet doesn't have valid serving certs
```

Comment:
If the kubelet doesn't have valid serving certs, would we still be interested in analyzing `iptables-operation-not-permitted` and other flaky stuff from that run?  Because I'd expect top-level test-cases for "all the kubelets have valid serving certs", and that investigating any failures there would be high-priority.  If that's true, this softening here doesn't get you access to much more data, and it does open you up to sharing your creds with a man-in-the-middle impersonator, and that seems like an ambiguous risk/reward trade-off.

Reply:
Ah, ok.  Backing that up with [docs][1]:

> This means the apiserver cannot verify the log data it is receiving came from the real kubelet.

and I agree that that's an acceptable risk :+1:

[1]: https://github.com/kubernetes/api/blob/ec59454f24f514bb472628bc0a5c8c9125218dbc/core/v1/types.go#L5776